src/TestLogger.php

```diff
@@ -65,6 +65,27 @@ final class TestLogger extends AbstractLogger
     /** @var array<int|string, array<int, array<string, mixed>>> */
     public array $recordsByLevel = [];
 
+    /** @var array<string, string|int> */
+    public array $levelMap = [
+        LogLevel::EMERGENCY => LogLevel::EMERGENCY,
+        LogLevel::ALERT => LogLevel::ALERT,
+        LogLevel::CRITICAL => LogLevel::CRITICAL,
+        LogLevel::ERROR => LogLevel::ERROR,
+        LogLevel::WARNING => LogLevel::WARNING,
+        LogLevel::NOTICE => LogLevel::NOTICE,
+        LogLevel::INFO => LogLevel::INFO,
+        LogLevel::DEBUG => LogLevel::DEBUG,
+    ];
+
+    /**
+     * @param array<string, string|int>|null $levelMap
+     *   Keys are LogLevel::*, values are alternative strings or integers used as log levels in the SUT.
```

Comment:
We can type-hint the allowed keys to phpstan like this:

```suggestion
     * @param array<LogLevel::*, string|int>|null $levelMap
     *   Keys are LogLevel::*, values are alternative strings or integers used as log levels in the SUT.
```